tests/allatonce/test_allatonceform.py

```diff
@@ -231,3 +237,72 @@ def form_mass(u, p, v, q):
             uparallel = aaoform.F.get_component(step, cpt)
             err = fd.errornorm(userial, uparallel)
             assert (err < 1e-12)
+
+
+@pytest.mark.parallel(nprocs=4)
+def test_time_update():
```

Comment:
This test seems to be more complicated than it needs to be. I think we just need to check:

- Does the `AllAtOnceForm` initialise `t0` and `time` correctly when created?
- Does `time_update()` with no arguments correctly shuffle `t0` and `time` along to the next window?
- Does `time_update(t=t)` correctly set `t0` to `t` and update `time` accordingly (we probably want to allow `t` to be either a float/int or a `Constant` so we should test both of these cases.

Please can you try and simplify this?

Reply:
Test looks much better now, thanks!